### PR TITLE
 Add OapOrcUtils to fix #935

### LINKED ISSUE #935: [OAP-0.5]All Parquet Query can't run with Spark-2.1.1
#843 Introduce UTF8String.java from spark-2.3, this class change signature of toXXX method(like toInt or toLong) between spark-2.1.0 to spark-2.3.0

- spark-2.1.0 no these methods

- after SPARK-19178(spark-2.1.1)  add `UTF8String.toXXX` methods and signature as following:
`int toInt() ` there is no args of these methods

- after SPARK-19843 (spark-2.2.0) `UTF8String.toXXX` methods changed signature as following:
`boolean toInt(IntWrapper intWrapper) `

Run all parquet query with oap 0.5 and spark-2.1.1 will trigger `java.lang.NoSuchMethodError: org.apache.spark.unsafe.types.UTF8String.toInt()I` problem,  there is no problem before #843.

Spark-2.1.2 & Spark-2.1.3 has same signature of `UTF8String.toXXX` method, so these versions should also have problems.

I don't know if there will be any other problems we run spark-2.1.x  + oap with `UTF8String.java` from spark-2.3, but I think we should fix this problems because oap 0.5 may the last version we support spark-2.1.x

### PR DESCRIPTION
**Pull request by zhixingheyi-tian**
## What changes were proposed in this pull request?

Add OapOrcUtils  instead of  importing UTF8String.java from spark2.3.
Fix #935

## How was this patch tested?

unit tests, integration tests

(If this patch involves UI changes, please attach a screenshot; otherwise, remove this)

### CODE REVIEW

File: src/main/java/org/apache/spark/sql/vectorized/oap/orc/OapOrcUtils.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+import static org.apache.spark.unsafe.Platform.*;
+
+/**
+ * Utilities to help manipulate classes for orc
+ */
+public class OapOrcUtils {
+
+    public static UTF8String copy(UTF8String utf8String)
+    {
```
Comment (LuciferYang) [inline src/main/java/org/apache/spark/sql/vectorized/oap/orc/OapOrcUtils.java]: need new newline?

File: src/main/java/org/apache/spark/sql/vectorized/oap/orc/OapOrcUtils.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.vectorized.oap.orc;
+
+import org.apache.spark.unsafe.types.UTF8String;
+
+import static org.apache.spark.unsafe.Platform.*;
```
Comment (LuciferYang) [inline src/main/java/org/apache/spark/sql/vectorized/oap/orc/OapOrcUtils.java]: only use Platform.copyMemory method, need static import？

File: src/main/java/org/apache/spark/sql/vectorized/oap/orc/OapOrcUtils.java [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+/**
+ * Utilities to help manipulate classes for orc
+ */
+public class OapOrcUtils {
+
+    public static UTF8String copy(UTF8String utf8String)
+    {
+        byte[] bytes = new byte[utf8String.numBytes()];
+        copyMemory(utf8String.getBaseObject(), utf8String.getBaseOffset(),
+                bytes, BYTE_ARRAY_OFFSET, utf8String.numBytes());
```
Comment (LuciferYang) [inline src/main/java/org/apache/spark/sql/vectorized/oap/orc/OapOrcUtils.java]: indent

**Comment by LuciferYang** [comment]
@zhixingheyi-tian on the other hand, may we can only introduce ColumnVector.java to Sprark-2.1 & Spark-2.2 do a few simple changes and remove all class under vectorized.oap.orc

File: src/main/java/org/apache/spark/sql/vectorized/oap/orc/OapOrcUtils.java (hunk 1 above)
Comment (zhixingheyi-tian) [reply src/main/java/org/apache/spark/sql/vectorized/oap/orc/OapOrcUtils.java]: Should be removed.

File: src/main/java/org/apache/spark/sql/vectorized/oap/orc/OapOrcUtils.java (hunk 2 above)
Comment (zhixingheyi-tian) [reply src/main/java/org/apache/spark/sql/vectorized/oap/orc/OapOrcUtils.java]: Both use "copyMemory " and "BYTE_ARRAY_OFFSET"

File: src/main/java/org/apache/spark/sql/vectorized/oap/orc/OapOrcUtils.java (hunk 3 above)
Comment (zhixingheyi-tian) [reply src/main/java/org/apache/spark/sql/vectorized/oap/orc/OapOrcUtils.java]: You are  right.

**Comment by LuciferYang** [comment]
Thx @zhixingheyi-tian fix this problem.

**Comment by LuciferYang** [comment]
This patch need back port to branch-0.5

**Pull request merged**